scrape/scrape.go

```diff
@@ -906,15 +910,15 @@ type scrapeCache struct {
 
 // metaEntry holds meta information about a metric.
 type metaEntry struct {
-	lastIter uint64 // Last scrape iteration the entry was observed at.
-	typ      textparse.MetricType
-	help     string
-	unit     string
+	metadata.Metadata
```

Comment:
This just embeds the previous fields. As Bryan mentioned, we can reuse the recently-introduced MetricType enum to avoid some back-and-forth conversions; this would impact the code in the current MetadataWatcher as well.
https://github.com/prometheus/prometheus/blob/00ec720c29bce637c1580a2ce43b32122d133f83/tsdb/record/record.go#L67-L79

I haven't gotten around to benchmarking the behavior, but I'll get to it.